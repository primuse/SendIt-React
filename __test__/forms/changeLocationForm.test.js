import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import LocationParcelForm, { LocationModalForm }
  from '../../src/components/forms/changeLocationForm.jsx';

const onChange = jest.fn();
const save = jest.fn();
const changeParcelLocation = jest.fn();
const props = {
  parcel: {
    isLoading: false
  }
};

const props2 = {
  parcel: {
    isLoading: true
  }
};

const mockStore = configureStore([thunk]);
const store = mockStore(props);
let wrapper;

describe('Change Parcel Location Form', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}>
        <LocationParcelForm />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Change Parcel Location Form', () => {
  beforeEach(() => {
    wrapper = mount(<LocationModalForm {...props}
      onChange={onChange} save={save}
      changeParcelLocation={changeParcelLocation} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('should change state', () => {
    const input = wrapper.find('#currentLocation');
    input.instance().value = 'USA';
    input.simulate('change');
    expect(wrapper.state().currentLocation).toEqual('USA');
  });
  it('should simulate the onSubmit', () => {
    wrapper.find('form').simulate('submit');
    const input1 = wrapper.find('input').at(0);
    input1.simulate('change', { target: { value: 'USA' } });
    expect(changeParcelLocation).toHaveBeenCalled();
  });
});

describe('Change Parcel Location Form', () => {
  beforeEach(() => {
    wrapper = mount(<LocationModalForm {...props2}
      onChange={onChange} save={save}
      changeParcelLocation={changeParcelLocation} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('should show the loading button', () => {
    expect(wrapper.find('span')).toBeDefined();
  });
});
