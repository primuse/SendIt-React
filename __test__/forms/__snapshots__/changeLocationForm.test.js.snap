// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Parcel Location Form renders correctly 1`] = `
<form
  className="text-right"
  id="locationForm"
  onSubmit={[Function]}
>
  <div
    className="form-row"
  >
    <input
      id="currentLocation"
      name="currentLocation"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <button
    className="btn xsm is-outlined white"
    type="submit"
  >
    Save
  </button>
</form>
`;
